addons/hr_expense/static/src/views/attachment_view.xml

```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<templates xml:space="preserve">
+    <t t-inherit="mail.AttachmentView" t-inherit-mode="extension">
+        <xpath expr="//h3/t" position="before">
+            <t t-if="state.thread.model === 'hr.expense.sheet'" t-out="`${state.thread.mainAttachment.res_name or state.thread.name} - `"/>
+        </xpath>
+    </t>
+</templates>
```

Comment:
I'm not fond of using xpaths in general, it's too dependent on the structure of the DOM, which makes it easy to break. All you are trying to achieve here is to change the text content of a node; xpaths are probably a bit overkill for this kind of purpose

My suggestion would be to define a getter on the AttachmentView component that returns the string to display. You can then `t-esc` the value of this getter in the template. Now, if you want to change the value, all you have to do is patch the getter

The advantage of doing things this way is that even if the structure of the DOM were to change completely, your patch would still work just fine